ui/perfherder/compare/CompareTable.jsx

```diff
@@ -33,7 +33,7 @@ export default class CompareTable extends React.PureComponent {
     return (
       <Table sz="small" className="compare-table mb-0 px-0" key={testName}>
         <thead>
-          <tr className="subtest-header bg-lightgray">
+          <tr id={testName.replace(/ /g, '-')} className="subtest-header bg-lightgray">
```

Comment:
I agree we should use shorter id names.
We could use signature ids for the specific tests, but we will need another way of specifying ids for header names. Header names can't be associated to a particular signature.

I believe the best solution would be to implement a custom hash function in Javascript, such as [this one](https://stackoverflow.com/questions/6122571/simple-non-secure-hash-function-for-javascript). We will feed it with the hyphen-based ids we already & get instead a shorter id.

We will use the same approach for both tests & header titles.

Reply:
Also agree with the icon hyperlink. I this this SO question explains [the technique we should employ](https://stackoverflow.com/questions/40217585/in-react-router-v4-how-does-one-link-to-a-fragment-identifier).

Reply:
Ah right, that's true. Each test has a unique signature id based on repository and framework. So it would matter if comparing against two different repositories, unless we find a way to implement it consistently. But yes, other options are fine too. Whatever is easiest.

Reply:
Regarding the hash function, wouldn't we have to store it?

Reply:
@sarah-clements if we're trying to deprecate the signature hash (I saw we actually store some hashes in json files IIRC) what we're going to use instead?
Also, how we store it is less important in my opinion (if I'm not missing something). I'm trying in the first place to make it work correctly

Reply:
> Regarding the hash function, wouldn't we have to store it?

If you're referring to storing the hash function in our code base, then yes, I suggest we add its implementation *(unless we already have one in our code base/dependencies)*.

Or maybe you're referring to storing the hash values on backend? If that's the question, I don't think we need that. The hash value is computed in a deterministic manner. It's basically the same thing as the name for a test, only in a shorter numerical form. And that test name is consistent across page reloads of the Compare view.

Reply:
> @sarah-clements if we're trying to deprecate the signature hash (I saw we actually store some hashes in json files IIRC) what we're going to use instead?

We already use the signature id, which is just a unique number generated by our database.